File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit/AnnotationOnScreenViewController.m
//
//  AnnotationOnScreenViewController.m
//
//  Copyright © 2016 Tokbox, Inc. All rights reserved.
//

#import "AnnotationOnScreenViewController.h"
#import <OTAnnotationKit/OTAnnotationKit.h>

@interface AnnotationOnScreenViewController()
@property (nonatomic) OTFullScreenAnnotationViewController *annotationOverContentViewController;
@end

@implementation AnnotationOnScreenViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.annotationOverContentViewController = [[OTFullScreenAnnotationViewController alloc] init];
    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Annotate" style:UIBarButtonItemStylePlain target:self action:@selector(startAnnotation)];
    
    UIButton *statusButton =  [[UIButton alloc] initWithFrame:CGRectMake(0, 0, CGRectGetWidth([UIScreen mainScreen].bounds), 20)];
    [statusButton addTarget:self action:@selector(stopAnnotation) forControlEvents:UIControlEventTouchUpInside];
    [statusButton setBackgroundColor:[UIColor greenColor]];
    [statusButton setTitle:@"You are annotating your screen, Tap here to dismiss" forState:UIControlStateNormal];
    [statusButton.titleLabel setFont:[UIFont systemFontOfSize:12.0f]];
    [self.annotationOverContentViewController.view addSubview:statusButton];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear: animated];
    
    CGRect navigationBarFrame = self.navigationController.navigationBar.frame;
    CGRect newFrame = CGRectMake(navigationBarFrame.origin.x, navigationBarFrame.origin.y, navigationBarFrame.size.width, 44);
    self.navigationController.navigationBar.frame = newFrame;
}

- (IBAction)viewPressed:(id)sender {
    [self.view bringSubviewToFront:sender];
}

- (void)startAnnotation {
    
    [[UIApplication sharedApplication] setStatusBarHidden:YES];
    CGRect navigationBarFrame = self.navigationController.navigationBar.frame;
    CGRect newFrame = CGRectMake(navigationBarFrame.origin.x, navigationBarFrame.origin.y, navigationBarFrame.size.width, 64);
    self.navigationController.navigationBar.frame = newFrame;
    [self presentViewController:self.annotationOverContentViewController animated:YES completion:nil];
}

- (void)stopAnnotation {
    [[UIApplication sharedApplication] setStatusBarHidden:NO];
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit/AnnotationOnScreenViewController.swift
//
//  AnnotationOnScreenViewController.swift
//
//  Copyright © 2016 Tokbox, Inc. All rights reserved.
//

import UIKit

class AnnotationOnScreenViewController: UIViewController {
    private let annotationOverContentViewController = OTFullScreenAnnotationViewController()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Annotate", style: .Plain, target: self, action: #selector(AnnotationOnScreenViewController.startAnnotation))
        
        let statusButton = UIButton(frame: CGRectMake(0, 0, CGRectGetWidth(UIScreen.mainScreen().bounds), 20))
        statusButton.addTarget(self, action: #selector(AnnotationOnScreenViewController.stopAnnotation), forControlEvents: .TouchUpInside)
        statusButton.backgroundColor = UIColor(red: 118.0/255.0, green: 206.0/255.0, blue: 31.0/255.0, alpha: 1.0)
        statusButton.setTitle("You are annotating your screen, Tap here to dismiss", forState: .Normal)
        statusButton.titleLabel!.font = UIFont.systemFontOfSize(12.0)
        annotationOverContentViewController.view.addSubview(statusButton)
    }
    
    @IBAction func viewPressed(sender: UIView!) {
        self.view.bringSubviewToFront(sender)
    }
    
    
    func startAnnotation() {
        
        UIApplication.sharedApplication().statusBarHidden = true
        let navigationBarFrame = navigationController!.navigationBar.frame
        let newFrame = CGRectMake(navigationBarFrame.origin.x, navigationBarFrame.origin.y, navigationBarFrame.size.width, 64)
        navigationController!.navigationBar.frame = newFrame
        self.presentViewController(annotationOverContentViewController, animated: true, completion: nil)
    }
    
    func stopAnnotation() {
        UIApplication.sharedApplication().statusBarHidden = false
        self.dismissViewControllerAnimated(true, completion: nil)
    }
}
